api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -50,6 +49,12 @@
    * <p>
    * Files that may contain some rows that match the expression and some rows that do not will
    * result in a {@link ValidationException}.
+   * <p>
+   * This has no requirements for the latest snapshot and will not fail based on other snapshot
+   * changes.
+   * <p>
+   * Note that either this method or {@link OverwriteFiles#deleteFile(DataFile)}
+   * should be used to mark deleted files, not both.
```

Comment:
This update isn't quite true. There are methods to determine which files to delete, and there are separate ways to configure validations.

When this operation is used to produce an idempotent overwrite, where it can be completed regardless of what files are added after it starts, it doesn't matter that the files were matched by a filter or directly by `deleteFile`. For example, I might want to overwrite an old file with new values for a column that was added. That overwrite is idempotent, but might use `deleteFile`.